From 0deb742260a09067cae495a630683986abbdedb3 Mon Sep 17 00:00:00 2001
From: Kevin Hao <haokexin@gmail.com>
Date: Thu, 10 Oct 2019 21:42:21 +0800
Subject: [PATCH] mmc: cavium-thunderx: Drop the IRQF_NO_THREAD constraint

The IRQF_NO_THREAD is added by a Marvell SDK patch 238e623e1024 ("mmc:
cavium: fix swiotlb buffer is full") in order to get back some of the
performance loss. But in some cases (such as rt kernel), we do need the
capability to thread irq handler. Otherwise we would get warnings because
the normal spin lock is used in the irq handler. So drop this constraint.

Signed-off-by: Kevin Hao <haokexin@gmail.com>
Signed-off-by: Bruce Ashfield <bruce.ashfield@gmail.com>
---
 drivers/mmc/host/cavium-thunderx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mmc/host/cavium-thunderx.c b/drivers/mmc/host/cavium-thunderx.c
index 87e10da5f627..c75535504447 100644
--- a/drivers/mmc/host/cavium-thunderx.c
+++ b/drivers/mmc/host/cavium-thunderx.c
@@ -49,7 +49,7 @@ static int thunder_mmc_register_interrupts(struct cvm_mmc_host *host,
 	/* register interrupts */
 	for (i = 0; i < nvec; i++) {
 		ret = devm_request_irq(&pdev->dev, pci_irq_vector(pdev, i),
-				       cvm_mmc_interrupt, IRQF_NO_THREAD,
+				       cvm_mmc_interrupt, 0,
 				       cvm_mmc_irq_names[i], host);
 		if (ret)
 			return ret;
-- 
2.14.5

